error_handler.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2020 The LevelDB-Go and Pebble Authors. All rights reserved. Use
+// of this source code is governed by a BSD-style license that can be found in
+// the LICENSE file.
+
+package pebble
+
+import (
+	"os"
+	"sync"
+
+	"github.com/cockroachdb/errors"
+	"github.com/cockroachdb/pebble/internal/base"
+	"github.com/cockroachdb/pebble/vfs"
+)
+
+// BackgroundErrorReason is an enum of possible operations whose failure may
+// put the DB in read-only mode depending on the severity.
+type BackgroundErrorReason uint8
```

Comment:
Should we call this `DBErrorReason`? Similary call `BackgroundError` as `DBError`? Calling them background sounds like they only deal with flush, compaction. But they cover user writes as well.